### PR TITLE
Implement SystemExit

### PR DESCRIPTION
**Pull request by orf**
I'm not sure how best to structure this. CPython throws a `SystemExit` when `exit()` (technically `sys.exit()`) is called, and this appears to be caught and handled here: https://github.com/python/cpython/blob/dcfcd146f8e6fc5c2fc16a4c192a0c5f5ca8c53c/Python/pythonrun.c#L650

I'm not sure that the `run_code_obj` is the best place to handle this, as the return value is just a `PyResult`. So callers would need to inspect to see if it's an exception type and check which seems iffy.

We might want a custom enum for unrecoverable 'shutdown-esque' errors, like `SystemExit` or `MemoryError`, so the code calling the VM can handle this appropriately?

### CODE REVIEW

File: vm/src/builtins.rs [hunk 1]
```diff
@@ -724,6 +724,11 @@ fn builtin_zip(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.ctx.new_list(new_items))
 }
 
+fn builtin_exit(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
```
Comment (tekknolagi) [inline vm/src/builtins.rs]: You're going to want to raise the exception with the argument to the exit function as the exception argument. Then, later, convert it to the exit code.

**Comment by orf** [comment]
I've refactored this to provide a `code` attribute on the SystemExit exception @tekknolagi. The shell now inspects the raised exception, and if it's a SystemExit then it exits with the right code.

I'm not too happy with how this is structured. I feel like we need a better way to define these exceptions to reduce boilerplate, but I'm not sure how we could do this.

**Comment by windelbouwman** [comment]
@orf do you wish to keep this PR? I think we still require this work in some form.

**Comment by windelbouwman** [comment]
Closing this PR since it's getting behind master too far.

**Pull request closed without merging**